cli_tools/gce_ovf_import/gce_utils/machine_type_provider.go

```diff
@@ -0,0 +1,108 @@
+//  Copyright 2019 Google Inc. All Rights Reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License
+
+package ovfgceutils
+
+import (
+	"fmt"
+	"github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/gce_ovf_import/ovf_utils"
+	daisycompute "github.com/GoogleCloudPlatform/compute-image-tools/daisy/compute"
+	"github.com/vmware/govmomi/ovf"
+	"google.golang.org/api/compute/v1"
+	"sort"
+)
+
+// MachineTypeProvider is responsible for providing GCE machine type based virtual appliance
+// properties defined in OVF descriptor
+type MachineTypeProvider struct {
+	OvfDescriptor *ovf.Envelope
+	ComputeClient daisycompute.Client
+	Project       string
+	Zone          string
+}
+
+// GetMachineType returns machine type based on properties in OVF descriptor
+func (mp *MachineTypeProvider) GetMachineType() (string, error) {
+	virtualSystem, err := ovfutils.GetVirtualSystem(mp.OvfDescriptor)
+	if err != nil {
+		return "", err
+	}
+	virtualHardwareSection, err := ovfutils.GetVirtualHardwareSection(virtualSystem)
+	if err != nil {
+		return "", err
+	}
+	memoryMB, err := ovfutils.GetMemoryInMB(virtualHardwareSection)
+	if err != nil {
+		return "", err
+	}
+	cpuCount, err := ovfutils.GetNumberOfCPUs(virtualHardwareSection)
+	if err != nil {
+		return "", err
+	}
+	machineTypes, err := mp.ComputeClient.ListMachineTypes(mp.Project, mp.Zone)
```

Comment:
Just use custom machine type, no need to list then iterate, we dont expect crazy types like 1T mem and 1 cpu right?

Reply:
There are rules on what combinations of cpu/memory are allowed which would have to be implemented and cpu/memory rounded up to match those rules.

Also, custom machines are more expensive than predefined machine types. 

Without knowing what combinations of cpu/memory users will import most frequently, it's not clear which approach is better: lookup (current implementation) or custom machines. 

I think we can leave this for Alpha and have a proper analysis with more data on how to determine machine type.